### PR TITLE
Cache the pid and hostname globally, flatstr before writing

### PR DESCRIPTION
**Pull request by mcollina**
This put us on par with bole for basic and object benchmarks.

Inspired by https://github.com/rvagg/bole/blob/f3ba893c7f948f459615007767e2ffee519bc14c/bole.js#L13-L14.

^^ @rvagg good job!

### CODE REVIEW

**Comment by davidmarkclements** [comment]
Just need to update readme benchmarks?,

**Comment by mcollina** [comment]
Yours come out nicer (damn maxbook pro 15'')

**Comment by davidmarkclements** [comment]
added benchmarks, we actually be bole in every test - but *only just*

File: README.md [hunk 1]
```diff
@@ -72,52 +72,61 @@ This produces:
 <a name="benchmarks"></a>
 ## Benchmarks
 
-As far as we know, it is one of the fastest loggers in town:
+As far as we know Pino is fastest logger in town.
```
Comment (mcollina) [inline README.md]: I would leave this as before. The difference is so tiny, and the approach between bole and pino is essentially the same now.
20ms over 10000 log lines is hardly perceivable.

File: README.md [hunk 2]
```diff
@@ -72,52 +72,61 @@ This produces:
...
-benchLogLevelInterpolateExtra*10000: 2150.711ms
-benchPinoInterpolateExtra*10000: 706.112ms
-benchPinoUnsafeInterpolateExtra*10000: 674.391ms
-benchPinoExtremeInterpolateExtra*10000: 359.784ms
-benchPinoUnsafeExtremeInterpolateExtra*10000: 346.463ms
+benchDebugInterpolateExtra*10000: 640.001ms
+benchBunyanInterpolateExtra*10000: 2888.825ms
+benchWinstonInterpolateExtra*10000: 2616.285ms
+benchBoleInterpolateExtra*10000: 1313.470ms
+benchLogLevelInterpolateExtra*10000: 1487.610ms
```
Comment (mcollina) [inline README.md]: @rvagg you should use quick-format as well :D.

File: README.md (hunk 1 above)
Comment (davidmarkclements) [inline README.md]: Fair enough

**Pull request merged**